pyproject.toml

```diff
@@ -89,6 +89,7 @@ webtest = "*"
 parameterized = "^0.9.0"
 opentelemetry-test-utils = "^0.47b0"
 ruff = "*"
+pyfakefs = "^5.7.1"
```

Comment:
Probably not really a better option, but just checking if you considered writing the tests without needing to bring in an extra dependency?

Reply:
yeah, the other alternatives seemed like it'd be a bit more headache than it's worth.  Just asking the question before adding dependencies. :)

Reply:
Yeah, good call out.

Reply:
What's your opinion on using [pytest's tmp_path fixture](https://docs.pytest.org/en/stable/how-to/tmp_path.html)? This is my usually my preferred way to test functionality that depends on the filesystem. It has a lot of nice properties IMO:

* You can use all the regular filesystem operations.
* It doesn't rely on mocking of any kind.
* It handles cleanup of test files.
* It preserves a small number (last 2?) of test run files on disk, which is IMO very useful for inspecting them to understand what's happening or what went wrong.
* It's easy to drop a breakpoint anywhere in your test and inspect the real filesystem state.

I'm not against in-memory mocks for filesystems but generally don't see much benefit when `tmp_path` exists.

Reply:
I’m not familiar with that option, so I hadn’t considered it. But happy to take a look if we feel strongly that it's a better option here.

Does it make sense here when we are working with specific known file paths like `/var/local/secrets.json`? As in, we're not just testing various operations on arbitrary files, but rather files in specific locations on the filesystem. Where is the boundary for logic and configuration we really want to be under test here?

Reply:
No worries, I don't feel really strongly about it, but that is how I'd have written these so I figured it was worth asking for your thoughts.

I normally introduce an option to set the output directory and set that when calling it in tests. It feels a little less magical and easier to debug to me since there's no mocking and it's real filesystem operations that you can look at with all the regular filesystem tools you're used to. It has downsides too, of course.

Reply:
Gotcha, thanks for the notes! I'll take a closer look at it next time I'm working in this space.